salalib/shapemap.cpp

```diff
@@ -2638,8 +2638,8 @@ bool ShapeMap::read( istream& stream, int version, bool drawinglayer )
    for (int k = 0; k < count; k++) {
       int key;
       stream.read((char *) &key, sizeof(key));
-      int index = m_objects.add(key, SalaObject());
-      m_objects.value(index).read(stream,version);
+      m_objects.insert(std::make_pair(key, SalaObject()));

```

Comment:
Use the iterator returned by insert for further actions, see http://www.cplusplus.com/reference/map/map/insert/

Reply:
Fixed in next pull request